packages/connectivity/src/scp-cf/connectivity-service.ts

```diff
@@ -38,7 +39,28 @@ export async function addProxyConfigurationOnPrem(
     };
   }
 
-  const proxyConfiguration: ProxyConfiguration = {
+  // Get the proxy authorization token if it exists
+  const proxyAuthorizationToken =
+    destination.proxyConfiguration?.headers?.['Proxy-Authorization']?.match(
+      /^Bearer (.+)/
+    )?.[1];
+
+  let proxyConfiguration: ProxyConfiguration;
+  if (proxyAuthorizationToken) {
```

Comment:
[q] Sorry I don't get why you need to check this here. Isn't the token validity already being checked in serviceToken function? The cache will returned undefined if the access_token is expired and trigger a clientCredentialToken call. So the refresh will happen if needed.

Reply:
Make totally sense. Thanks for pointing this out.